baseapp/abci.go

```diff
@@ -266,7 +278,19 @@ func (app *BaseApp) PrepareProposal(req abci.RequestPrepareProposal) abci.Respon
 // Ref: https://github.com/cosmos/cosmos-sdk/blob/main/docs/architecture/adr-060-abci-1.0.md
 // Ref: https://github.com/tendermint/tendermint/blob/main/spec/abci/abci%2B%2B_basic_concepts.md
 func (app *BaseApp) ProcessProposal(req abci.RequestProcessProposal) abci.ResponseProcessProposal {
-	// TODO: Implement.
+	ctx := app.deliverState.ctx
```

Comment:
```suggestion
	ctx := app.checkState.ctx
```

I found out that for the second block deliverState is nil, and that causes the app to not produce any more blocks.
What I would like to know if it's okay to use checkState here and that the first produced block was just because the deliverState gets initialized during initGenesis.

cc: @alexanderbez

Reply:
`BeginBlock` and `InitChain` are the only places we call for `app.setDeliverState` and `Commit()` resets it to nil but re-initializes the checkState. So I think this must be the way